engines/account/app/workers/account/signup_worker.rb

```diff
@@ -0,0 +1,13 @@
+module Account
+  class SignupWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :low
+
+    def perform(signup_id)
+      signup = Signup.find_by(id: signup_id)
+
+      SignupMailer.confirm_email(signup).deliver
```

Comment:
@enricostano Why not just ActiveJob (and you select then a backend, it can be sidekiq in production, Q or suckerpunch in development...)?:
```ruby
# No special worker for asyncing mails
SignupMailer.confirm_email(signup).deliver_later(wait: 5.seconds)

# Specific workers for other tasks
class MyJob < ActiveJob::Base
  queue_as :low

  def perform(cats, dogs=[], birds: nil)
    # whatever
  end
end

MyJob.set(wait_until: 4.days.from_now).perform_later([my_cat], birds: [my_bird])
MyJob.perform_later([my_cat, your_cat], [enrico_s_dog], birds: [my_bird])
# even perform in-thread (classmethod `.perform` is delegated to an instance)
MyJob.perform([my_cat, your_cat], [enrico_s_dog], birds: [my_bird])
```

ActiveJob is even able to pass DB objects (they are serialized/deserialized transparently) between the app thread and the worker thread (wherever it is).